runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java

```diff
@@ -590,9 +592,28 @@ Color getHidden(NamedQuantity item, int gePrice, int haPrice, boolean isTradeabl
 		final boolean canBeHidden = gePrice > 0 || isTradeable || !config.dontHideUntradeables();
 		final boolean underGe = gePrice < config.getHideUnderValue();
 		final boolean underHa = haPrice < config.getHideUnderValue();
+		final PriceDisplayMode mode = config.getHideValueCalculation();
+		final boolean underPrice;
+
+		if (mode == PriceDisplayMode.GE)
+		{
+			underPrice = underGe;
+		}
+		else if (mode == PriceDisplayMode.HA)
+		{
+			underPrice = underHa;
+		}
+		else if (mode == PriceDisplayMode.BOTH)
+		{
+			underPrice = underGe && underHa;
+		}
+		else
+		{
+			underPrice = false;
+		}
```

Comment:
PR looks good to me as is, but here's a suggestion to get rid of the if else block:
```suggestion
                final boolean underGe = gePrice < config.getHideUnderValue();
		final boolean underHa = haPrice < config.getHideUnderValue();
		final boolean underPrice;
		
		switch (config.getHideValueCalculation())
		{
		    case PriceDisplayMode.GE:
		        underPrice = underGe;
		        break;
		    case PriceDisplayMode.HA:
		        underPrice = underHa;
		        break;
		    case PriceDisplayMode.BOTH:
		        underPrice = underGe && underHa;
		        break;
		    default:
		        underPrice = false;
		}
```